### ISSUE TITLE
Running as a daemon

### ISSUE THREAD

**Issue by nik736** [bensheldon/good_job#88]
First of all great job and thanks for this gem! I love the idea and how it works.

I just wanted to ask what the best way is to run is as a deamon? Is a -d flag planned?

Thanks!

**Comment by sj26** [comment]
Personally I think baking daemonization into every process is an anti-pattern. Being a good daemon is difficult. Doing it with a tool which solves these problems like systemd or upstart or launchd is ideal.

If you just want to run the process in the background, can you background it, or use something like `nohup`:

```
nohup good_job
```

**Comment by gadimbaylisahil** [comment]
Just leaving it here for legacy folks. Also would love a review xD.

/etc/init.d/good-job

```shell
APP_USER=your_user_here
APP_ROOT=path_to_your_app

# GoodJob PID
# This can be improved to detect multiple processes and destroy but I have other monitoring in place
# Finds PID of good_job executable
PID=$(ps U "$APP_USER" | grep "[/]good_job" | awk '{ print $1 }')

# Command to run good_job
CMD="cd $APP_ROOT && RAILS_ENV=sandbox GOOD_JOB_MAX_THREADS=5 GOOD_JOB_QUEUES=queue1,queue2 GOOD_JOB_POLL_INTERVAL=10 bundle exec good_job start"

# Function to kill pid if it's set, with signal as argument
sig () {
  [ -n "$PID" ] && kill -$1 $PID
}

run () {
  if [ "$(id -un)" = "$APP_USER" ]; then
    eval $1
  else
    su -c "$1" - $APP_USER
  fi
}

case "$1" in
start)
  sig 0 && echo >&2 "Already running, only a single good-job process should be run, and previous one before deploy should be turned off first." && exit 1
  run "$CMD" &
  ;;
stop)
  sig TERM && exit 0
  echo >&2 "good-job is already turned off"
  ;;
status)
  if [ -n "$PID" ]; then
    echo "$good-job is running (pid $PID)."
    exit 0
  else
    echo "$good-job is not running."
    exit 1
  fi
  ;;
status_without_fail_exit)
  if [ -n "$PID" ]; then
    echo "good-job is running (pid $PID)."
  else
    echo "good-job is not running."
  fi
  ;;
*)
  echo >&2 "Usage: $0 <start|stop|status|status_without_fail_exit>"
  exit 1
  ;;
esac
```

`deploy.rb` - Capistrano

```ruby
  desc "Restart Jobs"
  task :restart_jobs do
    on roles(:app), in: :sequence, wait: 5 do
      # Current GoodJob PID
      execute :sudo, "-u #{fetch(:application)} /etc/init.d/#{fetch(:application)}-good-job status_without_fail_exit"
      # TERM GoodJob process
      execute :sudo, "-u #{fetch(:application)} nohup /etc/init.d/#{fetch(:application)}-good-job stop"
      # Starts new GoodJob process
      execute :sudo, "-u #{fetch(:application)} nohup /etc/init.d/good-job start"
      # Current GoodJob PID
      execute :sudo, "-u #{fetch(:application)} /etc/init.d/good-job status"
    end
  end
```

**Comment by bensheldon** [comment]
This feature has been released in `v1.6.0` 🎉

**Comment by gadimbaylisahil** [comment]
> This feature has been released in `v1.6.0` tada

Awesome news, thank you!
